src/end.sol

```diff
@@ -412,7 +418,7 @@ contract End {
         require(debt == 0, "End/debt-not-zero");
         require(vat.dai(address(vow)) == 0, "End/surplus-not-zero");
         require(block.timestamp >= add(when, wait), "End/wait-not-finished");
-        debt = vat.debt();
+        debt = sub(vat.debt(), cure.total());
```

Comment:
Maybe a good defensive change to the logic would be:
```
debt = vat.debt() >= cure.total() ? sub(vat.debt(), cure.total()) : vat.debt();
```
?

I am a little conflicted on it b/c while it protects us against certain accidents (e.g. setting the `cure` for some source to an anomalously huge amount), there are still accidents that will break things, using `vat.debt()` as the fallback is unlikely to be correct, and it doesn't really protect from malicious actors (they can still e.g. make `total == debt`). So after going to the trouble of suggesting it maybe I've talked myself out of it already, but going to post anyway for others to consider (maybe it inspires some more useful safeguard e.g. in the Cure itself).